pkg/scheduler/internal/cache/cache.go

```diff
@@ -483,6 +485,32 @@ func (cache *schedulerCache) RemovePod(pod *v1.Pod) error {
 	return nil
 }
 
+func (cache *schedulerCache) RemoveAssumedPod(pod *v1.Pod) error {
+	key, err := schedulernodeinfo.GetPodKey(pod)
+	if err != nil {
+		return err
+	}
+
+	cache.mu.Lock()
+	defer cache.mu.Unlock()
+
+	currState, ok := cache.podStates[key]
+	switch {
+	case ok && cache.assumedPods[key]:
+		err := cache.removePod(currState.pod)
+		if err != nil {
+			return err
+		}
+		delete(cache.podStates, key)
+		delete(cache.assumedPods, key)
+	case !ok && !cache.assumedPods[key]:
+		return nil
+	default:
+		return fmt.Errorf("pod %v could not found in podStates or assumedPods", key)
```

Comment:
Either write "could not be found" or change the sentence around to, "could not find pod ..."

Even then, though, the error message is inaccurate. It's trying to say that the pod may have been in one but not the other, because it's possible that `ok` was true, but the key was not present in the `cache.assumedPods` map, or that `ok` was false, but the key _was_ present in the `cache.assumedPods` map.

Under what conditions is the pod unknown to `cache.podStates` but present in `cache.assumedPods`?